packages/material-ui/src/Dialog/Dialog.test.js

```diff
@@ -181,6 +181,46 @@ describe('<Dialog />', () => {
         'should not fire the onBackdropClick callback',
       );
     });
+
+    it(
+      'should not close if the dialog receives the mousedown and the ' +
+        'backdrop receives the mouseup',
+      () => {
+        const onBackdropClick = spy();
+        wrapper.setProps({ onBackdropClick });
+
+        const backdrop = wrapper.find('div');
+        const content = wrapper.find(Paper).childAt(0);
```

Comment:
```suggestion
        const content = wrapper.find('[role="dialog"]').childAt(0);
```
and then add that role to the `Paper`. Improved a11y as a nice side-effect.

In general we try to move away from `childAt`. It's way to brittle WRT to context, forwardRef and HOCs.